contrib/babelfishpg_tsql/src/session.c

```diff
@@ -51,6 +52,23 @@ get_cur_db_name(void)
 	return pstrdup(current_db_name);
 }
 
+void 
+set_cur_db_name_for_parallel_worker(const char* logical_db_name)
+{
+	int len = strlen(logical_db_name);
+
+	Assert(len <= MAX_BBF_NAMEDATALEND);
+
+	if(logical_db_name == NULL || !DbidIsValid(get_db_id(logical_db_name)))
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_DATABASE),
+				 errmsg("database \"%s\" does not exist", logical_db_name)));
```

Comment:
I think %s with NULL string would be an issue. You can use "" in case of NULL

Reply:
Updated